### DISCUSSION TITLE
when R.binary function is used ?

### DISCUSSION THREAD

**Discussion by coleea** [ramda/ramda#3372] [Q&A] [answered]
I understand R.Binary's spcification from https://ramdajs.com/docs/#binary . 

But I don't know what situation I use this for.  I think It was made because it's useful. 

It is good if someone give me an example.

Thank you.

**Comment by kedashoe** [comment, marked as answer]
Not very often, I would say :stuck_out_tongue: But basically, if you are calling a function that takes however many arguments, you can use `unary`, `binary`, `nAry` to create a new function that takes exactly how many arguments you want. An example with `unary`

```js
let xs = ["1", "2", "3"];
console.log(xs.map(parseInt)); // => [1, NaN, NaN]
console.log(xs.map(R.unary(parseInt))); // => [1, 2, 3]
```

**Reply by FernandoBasso** [reply]
Nice @kedashoe. Thanks for the example, which is a practical one indeed.

Let's see if maybe more people come up with practical/real life examples, maybe also one really using  `binary`.

**Reply by coleea** [reply]
thank you for explanation.

in summary, unary, binary, and nAry is used for function that's arity varies ?

for example, I made a function that receives 3 parameters. and 2nd, 3rd argument is optional

in this case, the function operates differently depends on the number of parameters.

R.binary is used for this situation.

If I am wrong, please point it out.

thank you again
